### ISSUE TITLE
Linux Default Keybind clashes with readline

### ISSUE THREAD

**Issue by cwegener** [raphamorim/rio#183]
The default keybinds clash with readline, preventing the readline functions from working when running a program inside rio that uses readline (e.g. emacs, or any shell)

Cheatsheet of important readline keyboard shortcuts: https://www.bigsmoke.us/readline/shortcuts

The specific keybinds that clash are:

- `Control + N`
- `Control + W`
- `Control + T`

This wouldn't be a big problem if these could be changed via configuration.

Should the default keybinds be changed to more sensible ones instead?

**EDIT**: Forgot to include the configuration settings that I tried out to disable the conflicting keybinds, but that configuration did not result in disabling the default keybinds:

```
[bindings]
keys = [
	{ key = "n", with = "control", action = "none" },
	{ key = "t", with = "control", action = "none" },
	{ key = "w", with = "control", action = "none" }
]
```

**Comment by raphamorim** [comment]
Hey @cwegener thank you for the issue, very good point with readline keyboard. Considering is Control + key I suppose you are using linux right?

I wonder if the best solution would be change the defaults for `control + shift`, like `n + cntrl + shift`.

> EDIT: Forgot to include the configuration settings that I tried out to disable the conflicting keybinds, but that configuration did not result in disabling the default keybinds:

Could you please try again with latest main, believe fixed the overwrite in the latest changes.

**Comment by cwegener** [comment]
Changing the modifier to `control + shift` sounds like a good approach. There could be some other possible clashes when using `control + shift`  (e.g. the default keybinds for popular Desktop Environments may possibly be using `control + shift` for some of their global functions), but I think those clashes are a lower priority than the clashes with readline.

Yes, this is for Linux.

Btw. The config snippet I was testing with actually wasn't getting parsed successfully by `rio` for some reason (not sure why). So, that might have been why I wasn't able to override the defaults with my configuration.

I'll try with the latest main and report back.

Oh. And the rendering is really fast. (It's my first time testing `rio` and I was pretty impressed with the rendering speed. It definitely feels faster than any other option I've tried on Wayland ... and I think I've tried them all ... Currently using `wayst` as my main terminal because it's the least sucky one of them all ... )

**Comment by cwegener** [comment]
Ah. `action="ReceiveChar"` is what I need. All good now.

Changing the default modifier on Linux to `control` + `shift` would still be a good idea, considering how commonly readline is used (it's pretty much used by every shell and every REPL that I can think of)

(I'm not sure if `5263a12` is strictly necessary, as I clearly didn't read the documentation closely enough. It pretty clearly says 'None' and not 'none' :joy: ... But that change is a nice quality of life improvement nevertheless)

**Comment by raphamorim** [comment]
> Ah. action="ReceiveChar" is what I need. All good now.

that’s good to know!

> Changing the default modifier on Linux to control + shift would still be a good idea, considering how commonly readline is used (it's pretty much used by every shell and every REPL that I can think of)

would you be keen to send a PR with this change? Need to update rio/src/screen/bindings and the documentation file

**Comment by cwegener** [comment]
> would you be keen to send a PR with this change?

I'll give it my best shot. I haven't really read any rust code before. How hard can it be. :rofl:

**Comment by raphamorim** [comment]
No problem at all, I am here to help :D

In these lines:

https://github.com/raphamorim/rio/blob/193616be31814b43b7f0e7339d098ae4c5a890e7/rio/src/screen/bindings/mod.rs#L873-L876

https://github.com/raphamorim/rio/blob/193616be31814b43b7f0e7339d098ae4c5a890e7/rio/src/screen/bindings/mod.rs#L879

You can add ` | ModifiersState::SHIFT` after `ModifiersState::CONTROL` and should be it by the rust side. Then need to update the documentation file

**Comment by cwegener** [comment]
All good. Wasn't too hard to read. I don't claim to have understood every single line that I read. :wink:

PR has been created now #187

**Comment by raphamorim** [comment]
Closed by #187
